src/server/game/AI/CoreAI/ConversationAI.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef TRINITY_CONVERSATIONAI_H
+#define TRINITY_CONVERSATIONAI_H
+
+#include "Define.h"
+#include "ObjectGuid.h"
+
+class Conversation;
+class Unit;
+class Player;
+
+class TC_GAME_API ConversationAI
+{
+        uint32 _scriptId;
+
+    protected:
+        Conversation* const convo;
+    public:
+        explicit ConversationAI(Conversation* c, uint32 scriptId = {});
```

Comment:
we can also make scriptId an optional being defaulted to nullopt rather than handling the default value maybe

Reply:
given std::optional has a bit of overhead and we dont really need scriptid 0 to be valid i'm keeping it as is (just like it is for all the other ais)